src/lexer.l

```diff
@@ -61,6 +62,7 @@ struct lexer_param;
 "label" { return LABEL; }
 "break" { return BREAK; }
 "__loc__" { return LOC; }
+">|" { return HIGHPRECPIPE; }
```

Comment:
What is this used for? An example and a test would be great.

Reply:
It's about avoiding the need for parenthesis.  I'm not yet set on the operator... maybe `<|>`?

Reply:
How about `|>`?

Reply:
Let's see it in the context of a real use case where the current pipe operator fails. 
The question is if this new high priority pipe will be used everywhere in place of the old one just because it covers more cases, and if so, we should probably make sure it's fluent and good looking

Reply:
Yeah, I was afraid using the both operators together may reduce readability. I'm also afraid why adding a new operator is related to dlopen, which seems to be the main topic of this pull request. I'm not a member of the maintainers but I think it would be desirable to split a pull request per features in general.

Reply:
True, this PR does seem a bit overloaded, but the problem is that most of the additional features depend on some new core functionality from this branch. I'm not sure how to properly handle these, maybe at least rebase so that all new features are separate commits at the top?

However, this specific one with pipe priority doesn't seem to have a dependency. @nicowilliams let's get it out and have a proper discussion of the user complaints, use cases, and potential syntax for the solution, WDYT?

Reply:
@leonid-s-usov: questions about surprising behavior that is fixed by adding parenthesis come up all the time in the IRC channel for jq.  One came up a few days ago.  I forget the form, but maybe it was something like `some_exp | [blah | foo, bar]`, and the fix was to make it `some_exp | [(blah | foo), bar]`.

@itchyny I've considered all of these:

- `<|` ("bind tightly to the thing on the left")
- `|>` ("bind tightly to the thing on the right")
- `<|>` ("bind these two things tightly")
- `>|<` (ditto)

and so on, and also

- `$|` (reminding me of the `$` operator in Haskell, though here we're going for highest precedence rather than lowest)
- `||` (no -- it looks too much like a logical-or operator)

What I do know is that adding parenthesis is tiresome, so I want this.  And users need it too, though they might not really understand how to use it (but so what, we can always try documenting it better, or even not at all).

Reply:
let's move this discussion to #2038 which you've opened.

Reply:
Yes, this one is completely separable.  It's only in this PR because I happened to have it checked out and did the work there.  Most of the other things are all related, indeed, the most fundamental ones being "filehandles" and C-coded module loading ("dlopen"), which together enable just about everything else, and of which the latter is the most separable.